sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -1367,7 +1367,7 @@ class CastChecks extends ExprChecks {
   val sparkStringSig: TypeSig = cpuNumeric + BOOLEAN + TIMESTAMP + DATE + CALENDAR + STRING +
       BINARY + GpuTypeShims.additionalTypesStringCanCastTo
 
-  val binaryChecks: TypeSig = BINARY
+  val binaryChecks: TypeSig = STRING + BINARY
```

Comment:
A comment here explaining why binary can be STRING would be nice.

Reply:
I am confused by this. Spark supports the cast. Why do we need to explain that.

Reply:
sorry, I misunderstood this.